texastribune/data-visuals-create
# Add improvements for CMS integration

Leaving this as a draft until we can confirm this is what we need for the CMS

### Changed
- `templates/__common__/config/tasks/graphics-meta.js:` Adds more keys to manifest.json output and simplified how it's logging those warning messages. Also makes it so graphics without a title won't output in the manifest file.
- `templates/graphic/app/index.html`,  `templates/graphic/app/static.html`, `templates/graphic/app/templates/base.html` - Adds caption because we're now using description as the field for screenreaders/accessibility
-  `templates/__common__/config/utils.js` - Adds a helper for console logging with colors
- `templates/graphic/graphics-meta.md` - Documents the new caption field and rules.
- `templates/graphic/project.config.js` - Updates the default tag to be more of a catch all. This is to account for the event that someone doesn't update those tags in the project config, which seemed to be the case for the first handful we've done. @mandicai, does that sound like the right solve to you? Should they default as blank instead?

## Test steps
1. Check out `graphics-meta-fixes` and run `your/path/to/data-visuals-create/bin/data-visuals-create graphic my-test-graphic`
2. In that new project, add some sample HTML to the index.html. Feel free to use something from an old project or just this [test HTML](https://gist.github.com/ashley-hebler/44632a2c7eef5064b4d2b602dd100b97) I've been using.
3. Run `npm run serve` to preview the graphic.
4. Run `npm run build`
5. Confirm you see `caption:` appearing in the manifest.json output now.
6. Confirm the other new keys are there: `lastBuildTime`, `bucket`, `folder`, `graphicURL`
7. Leave either `graphicTitle` or `data-title` blank in your apps .html file. Run `npm run build`  again and confirm the graphic does not show up in the manifest now.
